File: algorithms/1_easy/R/00190_reverse_bits/solution.rs
impl Solution {
    pub fn reverse_bits(n: i32) -> i32 {
        let mut res = 0_u32;
        let mut num = n as u32;

        for _ in 0..i32::BITS {
            res = (res << 1) | (num & 1);
            num >>= 1;
        }

        res as i32
    }
}

File: algorithms/1_easy/R/00190_reverse_bits/solution.cpp
constexpr size_t INT_BITS = sizeof(int) * CHAR_BIT;

class Solution {
public:
    int reverseBits(int n) {
        unsigned int res = 0u;
        unsigned int num = static_cast<unsigned int>(n);

        for (size_t i = 0; i < INT_BITS; i += 1) {
            res = (res << 1) | (num & 1);
            num >>= 1;
        }

        return (int) res;
    }
};
